src/main/java/org/opentripplanner/routing/algorithm/raptor/router/RaptorRouter.java

```diff
@@ -0,0 +1,127 @@
+import java.time.LocalDate;
+import java.time.ZoneId;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * Does a complete transit search, including access and egress legs.
+ */
+public class RaptorRouter {
+    private final TransitDataProvider<TripSchedule> otpRRDataProvider;
+    private final TransitLayer transitLayer;
+    private static final Logger LOG = LoggerFactory.getLogger(RaptorRouter.class);
+
+    private static final RangeRaptorService<TripSchedule> rangeRaptorService = new RangeRaptorService<>(
+            // TODO - Load turning parameters from config file
+            new TuningParameters() {}
+    );
+
+    //TODO Naming
+    public RaptorRouter(RoutingRequest request, TransitLayer transitLayer) {
+        double startTime = System.currentTimeMillis();
+        this.otpRRDataProvider = new RaptorRoutingRequestTransitData(
+                transitLayer, startDate(request),2, request.modes, request.walkSpeed
+        );
+        LOG.info("Filtering tripPatterns took {} ms", System.currentTimeMillis() - startTime);
+        this.transitLayer = transitLayer;
+    }
+
+    public TripPlan route(RoutingRequest request) {
+
+        /* Prepare access/egress transfers */
+
+        double startTimeAccessEgress = System.currentTimeMillis();
+
+        Map<Stop, Transfer> accessTransfers =
+            AccessEgressRouter.streetSearch(request, false, Integer.MAX_VALUE);
+        Map<Stop, Transfer> egressTransfers =
+            AccessEgressRouter.streetSearch(request, true, Integer.MAX_VALUE);
+
+        TransferToAccessEgressLegMapper accessEgressLegMapper = new TransferToAccessEgressLegMapper(transitLayer);
+
+        Collection<TransferLeg> accessTimes = accessEgressLegMapper.map(accessTransfers, request.walkSpeed);
+        Collection<TransferLeg> egressTimes = accessEgressLegMapper.map(egressTransfers, request.walkSpeed);
+
+        LOG.info("Access/egress routing took {} ms", System.currentTimeMillis() - startTimeAccessEgress);
+
+        /* Prepare transit search */
+
+        double startTimeRouting = System.currentTimeMillis();
+
+        // TODO Time zones
+        int departureTime = Instant.ofEpochMilli(request.dateTime * 1000).atZone(ZoneId.systemDefault()).toLocalTime().toSecondOfDay();
+
+        // TODO Expose parameters
+        // TODO Remove parameters from API
+        RequestBuilder builder = new RequestBuilder();
+        builder.profile(RangeRaptorProfile.MULTI_CRITERIA)
+                .searchParams()
+                .earliestDepartureTime(departureTime)
+                .searchWindowInSeconds(40  * 60)
+                .addAccessStops(accessTimes)
+                .addEgressStops(egressTimes)
+                .boardSlackInSeconds(request.boardSlack)
+                .timetableEnabled(false);
+
+        //TODO Check in combination with timetableEnabled
+        builder.enableOptimization(Optimization.PARETO_CHECK_AGAINST_DESTINATION);
```

Comment:
I'm not sure what the cause is here, but with this enabled itineraries consistently start at `earliestDepartureTime + searchWindowInSeconds` (instead of `earliestDepartureTime`).

Reply:
Setting a value for `latestArrivalTime` seems to fix this issue, which is caused by the reverse search heuristic running with a time of `-1`.

I'm not sure what a sensible default is: `earliestDepartureTIme + searchWindowInSeconds`? A separate `maxSearchTimeInSeconds` parameter?

Reply:
It looks like this was fixed in d6adca0. Marking resolved, but @flaktack let us know if the problem still needs attention.